compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -353,7 +358,7 @@ impl<'tcx> Body<'tcx> {
     /// Returns `true` if a cycle exists in the control-flow graph that is reachable from the
     /// `START_BLOCK`.
     pub fn is_cfg_cyclic(&self) -> bool {
-        graph::is_cyclic(self)
+        self.is_cyclic.is_cyclic(self)
```

Comment:
This needs to be invalidated whenever a terminator may change

Reply:
Thanks! Do you happen to have something in mind how I could cover that with a test?

Reply:
Not really, outside of maybe a unit test that just calls a bunch of functions and checks that the flag was invalidated?